### PR TITLE
Add NET_RAW capability on etcd container securitycontext

### PR DESCRIPTION
**Pull request by Dr-Shadow**
Related to https://github.com/piraeusdatastore/piraeus-operator/issues/204

Not tested under containerd but it should be working since containerd doesn't drop NET_RAW by default

The capability should be added on container that's why I'm not using the configurable value PodSecurityContext (see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#podsecuritycontext-v1-core)

### CODE REVIEW

**Comment by WanzenBug** [comment]
Thanks for getting to the bottom of this!

Would you be willing to improve this a bit further by:

* dropping all capabilities
* adding only those that are actually needed.

That should prevent similar issues in the future.
You might also need to run `make deploy/piraeus` once to sync the example deployment with your changes to the template.

As a side node: seems my test setup using kubespray overrides the cri-o default capabilities with their own, which is why I couldn't reproduce the issue.

**Comment by Dr-Shadow** [comment]
Another way to fix it would be to remove the ping commands there (so it means we have to find another way to check if pod is ready) :

https://github.com/piraeusdatastore/piraeus-operator/blob/master/charts/piraeus/charts/etcd/templates/statefulset.yaml#L183

https://github.com/piraeusdatastore/piraeus-operator/blob/master/charts/piraeus/charts/etcd/templates/statefulset.yaml#L191

So that the default capabilities should be enough.

I'm wondering why we aren't using some official etcd chart like the bitnami one ? https://artifacthub.io/packages/helm/bitnami/etcd

**Comment by WanzenBug** [comment]
> Another way to fix it would be to remove the ping commands there

I already had a look through the available commands in the "official" etcd image, and ping seemed to be the only reasonable. You might be able to use an init-container for the same purpose, but that badly interacts with the readiness state and DNS resolution I think. Would have to be tried

> I'm wondering why we aren't using some official etcd chart like the bitnami one

Reason is given in https://github.com/piraeusdatastore/piraeus-operator/commit/7c35509df81819d270f2788c74ca5dc8f7cbcbee
TLDR: we had issues that a simple node failure wasn't recoverable. The replacement seemed to solve those issue.

But I agree in principle. It would be easier if we didn't have to maintain an Etcd chart. There is some ongoing work in LINSTOR to directly use the K8s API as persistent storage (like calico for example). But that will take time, so until then we need Etcd.

**Comment by Dr-Shadow** [comment]
Ok since I would have to test at least these capabilities, I would just stick with the default ones provided by CRI-O / containerd unless we have to add specific ones.
Especially since I'm not an etcd expert, I can't tell if capabilities are used or not.

From `crio config --default`
```
# List of default capabilities for containers. If it is empty or commented out,
# only the capabilities defined in the containers json file by the user/kube
# will be added.
default_capabilities = [
        "CHOWN",
        "DAC_OVERRIDE",
        "FSETID",
        "FOWNER",
        "SETGID",
        "SETUID",
        "SETPCAP",
        "NET_BIND_SERVICE",
        "KILL",
]
```

**Review by WanzenBug — DISMISSED** [review]
Looks good to me, LGTM. 

Only thing missing is to run `make deploy/piraeus` once and commit those changes. Then CI should pass and we can merge this.

**Comment by Dr-Shadow** [comment]
> Only thing missing is to run `make deploy/piraeus` once and commit those changes. Then CI should pass and we can merge this.

Oh that's why the tests were failing, good to know

**Review by WanzenBug — APPROVED** [review]
